Add register to users table

dankmitchell opened the pull request:
### Context
No need to separate `users` and `download_users` in the database.

### Changes proposed in this pull request
- Add `register` attr to `users` table
- Migrate all date from `download_users` into `users`
- Update `download_users` controller to save to `users` table
- Fix specs

### Guidance to review
Running the migrations should copy the data from `download_users` into `users` and completing the download form and api key form on the frontend should results in the data being saved to the `users` table

Commit:
Add register column to users table

Commit:
Add migration to move data from download_users to users

Commit:
Update migration

Commit:
Update download_users controller to persist to users tables

Commit:
Fix lint error

gidsg commented:
would be nice to add a `user_type` column to make it easier to distinguish the users. This could be an enum (either Postgres or Rails int style).

gidsg committed:
add user type activerecord enum

gidsg committed:
lint

gidsg committed:
remove test

this was causing a warning see: https://www.rubydoc.info/github/thoughtbot/shoulda-matchers/master/Shoulda%2FMatchers%2FActiveModel:validate_inclusion_of

Commit:
Commit schema file

michaelabenyohai commented:
The migration doesn't populate user_type for existing download or API users

michaelabenyohai commented on db/migrate/20180612082738_move_download_users_to_users.rb:
```diff
@@ -0,0 +1,15 @@
+        department: download_user.department,
+        is_government: download_user.is_government,
+        api_key: nil,
+        contactable: false
```
Just curious why contactable is false for existing download users?

michaelabenyohai replied:
Does this also need to be populated for API users?

michaelabenyohai commented:
When giving my details after download it doesn't appear to populate the `non_gov_use_category` or `department` in the `users` table.

michaelabenyohai commented:
When trying to get an API key registers-selfservice returns a 500 error. Does anyone else get this?

gidsg commented:
@michaelabenyohai have you definitely run migrations on self-service. If so, what is the exception in the self-service logs?

gidsg commented:
@michaelabenyohai ... thinking more about it, you can get a 500 if you don't have`GOOGLE_CREDENTIALS_JSON` set.

michaelabenyohai commented:
Yep that was it @gidsg! Sorry forgot about that.

When I choose not government we don't populate the `contactable` column for API key users. I assume this is because we don't ask them. Is it worth us populating this with false? What does it mean if this column is nil?

gidsg commented:
@michaelabenyohai 
> What does it mean if this column is nil?

currently we would interpret it as `false` this is because it would apply to early users created prior to this field existing. We could clarify it in a subsequent migration though.

gidsg committed:
set contactable state based on time of wording change

dankmitchell commented on db/migrate/20180612082738_move_download_users_to_users.rb:
```diff
@@ -8,7 +8,7 @@ def change
         is_government: download_user.is_government,
         api_key: nil,
-        contactable: false
+        contactable: download_user.created_at.between?(Time.utc(2018, 0o5, 30, 18), Time.now)
```
Is this a typo @gidsg ? Should be `Time.utc(2018, 05, 30, 18)`

gidsg replied:
@dankmitchell linter made me do it m8 https://github.com/bbatsov/ruby-style-guide#numeric-literal-prefixes
